src/scripts/main.js

```diff
@@ -1,3 +1,341 @@
 'use strict';
 
-// write your code here
+const everySquare = Array.from(document.querySelectorAll('.field-cell'));
+const everyRow = Array.from(document.querySelectorAll('.field-row'));
+const mainButton = document.querySelector('.start');
+const gameScore = document.querySelector('.game-score');
+let canMoveInAnyDirection = [];
+let score = 0;
+
+function appendNumber() {
+  const emptySquares = everySquare.filter(square => square.innerHTML === '');
+
+  if (emptySquares.length === 0) {
+    return;
+  }
+
+  const index = Math.floor(Math.random() * emptySquares.length);
+  const randomSquare = emptySquares[index];
+  const value = Math.random() > 0.9 ? 4 : 2;
+
+  setCellValue(randomSquare, value);
+}
+
+function setColumn(filtered) {
+  return filtered.map(cell => parseInt(cell.innerHTML, 10) || '');
+}
+
+function setCellValue(cell, value) {
+  cell.innerHTML = value || '';
+  cell.className = 'field-cell' + (value ? ` field-cell--${value}` : '');
+}
+
+function getNewRow(column, direction) {
+  const filtered = column.filter(num => num);
+  const missing = 4 - filtered.length;
+  const unfilled = Array(missing).fill('');
+
+  return (
+    direction === 'down' || direction === 'right'
```

Comment:
it's better to move the direction ('down', 'right', 'up', 'left' ) to the enum
it's will make a code cleaner